component/src/main/java/io/siddhi/extension/io/gcs/sink/GCSSink.java

```diff
@@ -0,0 +1,334 @@
+     *                         get siddhi related utility functions.
+     * @return StateFactory for the Function which contains logic for the updated state based on arrived events.
+     */
+    @Override
+    protected StateFactory<GCSSinkState> init(StreamDefinition streamDefinition, OptionHolder optionHolder,
+                                              ConfigReader configReader,
+                                              SiddhiAppContext siddhiAppContext) {
+        this.gcsSinkConfig = new GCSSinkConfig(optionHolder, siddhiAppContext.getScheduledExecutorService());
+        this.eventPublisher = new EventPublisher(gcsSinkConfig, optionHolder);
+
+        return GCSSinkState::new;
+    }
+
+
+    /**
+     * This method will be called when events need to be published via this sink
+     *
+     * @param payload        payload of the event based on the supported event class exported by the extensions
+     * @param dynamicOptions holds the dynamic options of this sink and Use this object to obtain dynamic options.
+     * @param state          current state of the sink
+     * @throws ConnectionUnavailableException if end point is unavailable the ConnectionUnavailableException thrown
+     *                                        such that the  system will take care retrying for connection
+     */
+    @Override
+    public void publish(Object payload, DynamicOptions dynamicOptions, GCSSinkState state)
+            throws ConnectionUnavailableException {
+        eventPublisher.publish(payload, dynamicOptions);
+    }
+
+
+    /**
+     * This method will be called before the processing method.
+     * Intention to establish connection to publish event.
+     *
+     * @throws ConnectionUnavailableException if end point is unavailable the ConnectionUnavailableException thrown
+     *                                        such that the  system will take care retrying for connection
+     */
+    @Override
+    public void connect() throws ConnectionUnavailableException {
+        this.gcsSinkConfig.setMapType(getMapper().getType());
+        eventPublisher.initializeServiceClient();
+    }
+
+    /**
+     * Called after all publishing is done, or when {@link ConnectionUnavailableException} is thrown
+     * Implementation of this method should contain the steps needed to disconnect from the sink.
+     */
+    @Override
+    public void disconnect() {
+
+    }
+
+    /**
+     * The method can be called when removing an event receiver.
+     * The cleanups that have to be done after removing the receiver could be done here.
+     */
+    @Override
+    public void destroy() {
+        this.eventPublisher =  null;
+    }
+
+    /**
+     * Returns the map type of a given Stream definition of a Sink
+     * @param streamDefinition
+     * @return
+     */
+    private String extractMapType(StreamDefinition streamDefinition) {
+        Optional<Annotation> mapAnnotation = streamDefinition.getAnnotations()
```

Comment:
instead of extracting mapType from stream definition, get it from Sink.getMapper()